### PR TITLE
Added city exists validator

### PR DESCRIPTION
**Pull request by rogergros**
Added city exists validator for city field on address

Ping @elcodi/owners

### CODE REVIEW

File: src/Elcodi/Bundle/GeoBundle/Resources/config/validation.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Elcodi\Component\Geo\Entity\Address:
+    properties:
+        city:
+            - Elcodi\Component\Geo\Validator\Constraints\CityExists: ~
```
Comment (xphere) [inline src/Elcodi/Bundle/GeoBundle/Resources/config/validation.yml]: It does not work with `city_exists`?

**Comment by xphere** [comment]
:+1:

File: src/Elcodi/Bundle/GeoBundle/Resources/config/validation.yml (hunk 1 above)
Comment (rogergros) [inline src/Elcodi/Bundle/GeoBundle/Resources/config/validation.yml]: Nope, city_exists is the tag for the constraint validator, and we need here the constraint :)

**Comment by rogergros** [comment]
Hey @elcodi/owners if no other comment is added, I'll merge this today at noon :)

File: src/Elcodi/Component/Geo/Validator/Constraints/CityExistsValidator.php [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+ * Class CityExistsValidator
+ */
+class CityExistsValidator extends ConstraintValidator
+{
+    /**
+     * @var LocationProviderInterface
+     *
+     * A location provider
+     */
+    private $locationProvider;
```
Comment (alch) [inline src/Elcodi/Component/Geo/Validator/Constraints/CityExistsValidator.php]: We are generally using `protected` accessors in Elcodi services to maximize  *open/closed*-ness

File: src/Elcodi/Component/Geo/Validator/Constraints/CityExistsValidator.php [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        try {
+            $location = $this->locationProvider->getLocation(
+                $value
+            );
+        } catch (EntityNotFoundException $e) {
+            $location = null;
+        }
+
+        if (
+            empty($location) ||
```
Comment (alch) [inline src/Elcodi/Component/Geo/Validator/Constraints/CityExistsValidator.php]: Wouldn't it safer to use a `is_null` here?

File: src/Elcodi/Component/Geo/Validator/Constraints/CityExistsValidator.php (hunk 2 above)
Comment (rogergros) [inline src/Elcodi/Component/Geo/Validator/Constraints/CityExistsValidator.php]: You're right, fixing it

File: src/Elcodi/Component/Geo/Validator/Constraints/CityExistsValidator.php (hunk 3 above)
Comment (rogergros) [inline src/Elcodi/Component/Geo/Validator/Constraints/CityExistsValidator.php]: Well, that would be better :)

**Comment by rogergros** [comment]
Master rebased

**Pull request merged**